### ISSUE TITLE
Grails 3.1.0.M3 is trying is read my maven settings.xml and failing to create app

### ISSUE THREAD

**Issue by bond-** [grails/grails-core#9487]
Grails 3.1.0.M3 was reading my settings.xml present in \<user.home\>/.m2/settings.xml

Stacktrace:
```
| Error Error occurred running Grails CLI: Failed to determine active profiles:
    Failed to determine activation for profile snc-release at null
 (NOTE: Stack trace has been filtered. Use --verbose to see entire trace.)
java.lang.IllegalStateException: Failed to determine active profiles:
    Failed to determine activation for profile snc-release at null

	at org.springframework.boot.cli.compiler.maven.MavenSettings.determineActiveProfiles(MavenSettings.java:134)
	at org.springframework.boot.cli.compiler.maven.MavenSettings.<init>(MavenSettings.java:87)
	at org.springframework.boot.cli.compiler.maven.MavenSettingsReader.readSettings(MavenSettingsReader.java:64)
	at org.springframework.boot.cli.compiler.grape.SettingsXmlRepositorySystemSessionAutoConfiguration.getSettings(SettingsXmlRepositorySystemSessionAutoConfiguration.java:47)
	at org.springframework.boot.cli.compiler.grape.SettingsXmlRepositorySystemSessionAutoConfiguration.apply(SettingsXmlRepositorySystemSessionAutoConfiguration.java:38)
	at org.springframework.boot.cli.compiler.grape.AetherGrapeEngineFactory.create(AetherGrapeEngineFactory.java:59)
	at org.grails.cli.profile.repository.MavenProfileRepository.<init>(MavenProfileRepository.groovy:51)
	at org.grails.cli.GrailsCli.createMavenProfileRepository(GrailsCli.groovy:327)
	at org.grails.cli.GrailsCli.execute(GrailsCli.groovy:236)
	at org.grails.cli.GrailsCli.main(GrailsCli.groovy:162)
| Error Error occurred running Grails CLI: Failed to determine active profiles:
    Failed to determine activation for profile snc-release at null
```

Workaround:
Delete my old maven ~/.m2/settings.xml
Bad workaround but only solution to me right now.

**Comment by graemerocher** [comment]
Grails 3.1 uses Aether to resolve profiles for app creation hence why it reads `settings.xml`, I guess the question is why it is causing this error

**Comment by pledbrook** [comment]
What profile activation conditions do you have in your _settings.xml_? I'd guess one of those is causing the issue based on the error message.

**Comment by bond-** [comment]
I'm having:
```
<profile>
	<id>snc-release</id>
	<activation>
		<file>
			<exists>${user.home}/.m2/some_dumy_file</exists>
		</file>
	</activation>

	<repositories>
		<repository>
		</repository>
	</repositories>
</profile>
```

**Comment by pledbrook** [comment]
I don't really know what's going on here, but it seems like the file activation is broken. Using `<jdk>` or `<property>` activation works fine (or at least doesn't cause an exception). Since Spring Boot is the one doing the parsing of _settings.xml_, I suggest throwing this issue to its team.

It definitely looks like they're [adding the `FileProfileActivation`](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-cli/src/main/java/org/springframework/boot/cli/compiler/maven/MavenSettings.java#L158), so I don't understand why this wouldn't work.

**Comment by graemerocher** [comment]
Please report the issue to https://github.com/spring-projects/spring-boot/issues and link to this issue for reference

**Comment by bond-** [comment]
Done. https://github.com/spring-projects/spring-boot/issues/4826

**Comment by bond-** [comment]
This is the same as: #9315

**Comment by bond-** [comment]
@graemerocher The third party issue has been fixed in version 1.3.2 of spring-boot
